### PR TITLE
1967/dust off docs

### PR DESCRIPTION
**Pull request by laclaro**
The pull request includes the restructuring of the first parts of the documentation.
 It now includes a dedicated WebUI chapter to introduce the UI and link to the detailed information.

    * 4. WebUI
        * Tokens
        * Users
        * Machines
        * Config
        * Audit
        * Components

Build the docs with `make htmltodo` to include `.. todo::` comments.

### CODE REVIEW

File: doc/tokens/tokentypes/pw.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.. _pw_token:
+
+Lost Password Token (pw)
```
Comment (cornelinux) [inline doc/tokens/tokentypes/pw.rst]: Please change this to "Password Token".

File: doc/tokens/tokentypes/questionnaire.rst [hunk 2]
```diff
@@ -8,15 +8,21 @@ Questionnaire Token
...
-many questions as specified with answers only he knows.
+During enrollment of such a *questionnaire* type token, the user answers at least as
+many questions as specified by the administrator with answers only he knows.
+
+.. figure:: images/enroll_questionnaire.png
+   :width: 500
 
 This token is a challenge response token.
-During authentication the user must give the token PIN and the a random
+During authentication the user gives the token PIN and a random
```
Comment (cornelinux) [inline doc/tokens/tokentypes/questionnaire.rst]: During authentication the user first gives the token PIN and the he is presented with a random question to which he gives the answer only he knows.

File: doc/tokens/tokentypes/questionnaire.rst [hunk 3]
```diff
@@ -8,15 +8,21 @@ Questionnaire Token
...
 question from the answered question is chosen. The user has to answer with
 the same answer he defined earlier.
 
-.. note:: If the administrator changes the questions _after_ a token was
+.. note:: By default, no questions are defined, so the administrator has to setup those
+   in "Config->Tokens->Questionnaire" before a questionnaire token can be rolled out successfully.
```
Comment (cornelinux) [inline doc/tokens/tokentypes/questionnaire.rst]: Link to config

File: doc/webui/index.rst [hunk 4]
```diff
@@ -0,0 +1,124 @@
+.. _webui:
+
+WebUI
+=========
+
+privacyIDEA comes with a web-based user interface which is used to manage and configure
+the privacyIDEA server. With the correct policies, it is also used a self-service portal
```
Comment (cornelinux) [inline doc/webui/index.rst]: Administrators can log in to configure the system and manage other users tokens.
Normal users can login and manage their own tokens.

Comment (cornelinux) [reply doc/webui/index.rst]: Please remove "with the correct policies".

File: doc/webui/index.rst [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+for the average user. This section gives an overview on the interface and links the
+respective sections in the documentation.
+
+.. note:: For productive environments, it is absolutely necessary to secure the connection with HTTPS.
```
Comment (cornelinux) [inline doc/webui/index.rst]: I would not note this.

File: doc/webui/index.rst [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+   :maxdepth: 1
+
+   user_details
+   manage_users
+
+
+Machines
+-------
+
+.. todo:: Short sentence about the purpose of machine-resolving.
```
Comment (cornelinux) [inline doc/webui/index.rst]: Machines are only necessary if you plan special use cases like managing SSH keys or doing offline OTP.
In most cases there is no need to manage machines.

File: doc/webui/index.rst [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+.. figure:: ../audit/auditlog.png
+   :width: 500
+
+    *Events can be displayed in the Audit log.*
+
+.. _components:
+
+Components
+-------
```
Comment (cornelinux) [inline doc/webui/index.rst]: too short :-)

File: doc/webui/index.rst [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+.. index:: Components
+
+Starting with privacyIDEA 2.15 you can see privacyIDEA components in the Web UI.
+privacyIDEA collects authenticating clients with their User Agent. Usually
+this is a type like *PAM*, *FreeRADIUS*, *OTRS*, *Wordpress*...
```
Comment (cornelinux) [inline doc/webui/index.rst]: We could remove OTRS and add *ownCloud* and *privacyIDEA Credential Provider*.

File: doc/installation/ubuntu.rst [hunk 9]
```diff
@@ -112,26 +105,13 @@ But we recommend installing the meta package::
...
-alternatively use the meta package ``privacyidea-nginx``.
-
-After installing with Apache2 or Nginx you only need to create your first
-administrator and you are done::
-
-   pi-manage admin add admin -e admin@localhost
+everything accordingly. Also in the ppa, the package ``privacyidea-nginx``
+is available.
 
+------------
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/ubuntu.rst]: should stay. The line separates from the sub headings.

File: doc/installation/upgrade.rst [hunk 10]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
...
 
+.. _upgrade_packaged:
+
+Upgrading a packaged installation
+...............
+
+In general, the upgrade of a packaged version of privacyIDEA should be done using the
+default tools (e.g. apt and yum). In any case, read the
+`READ_BEFORE_UPDATE <https://github.com/privacyidea/privacyidea/blob/master/READ_BEFORE_UPDATE.md>`_
+file. As there is a vast amount of possible custom installation cases, it cannot be guaranteed
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst [hunk 11]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
 
 Now you need to restart your webserver for the new code to take effect.
 
+.. _upgrade_packaged:
+
+Upgrading a packaged installation
+...............
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst [hunk 12]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
...
+Ubuntu upgrade
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you use the Ubuntu packages in a default setup, the upgrade can should be done
+using::
+
+   apt update
+   apt dist-upgrade
+
+In case you upgrade from the old privacyIDEA 2.23.x to the version 3.x you have to
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: Another subheadline is toooo small. I chose a note instead.

File: doc/installation/upgrade.rst [hunk 13]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
...
+in the repository instead come with a virtual python environmentThis may cause lots
+of obsolete packages after upgrading which may be removed with::
+
+   apt autoremove
+
+
+CentOS upgrade
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For a default community CentOS installation run::
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: changed to RHEL

File: doc/installation/upgrade.rst [hunk 14]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
...
+
+ yum update
+
+to upgrade.
 
 Upgrade to privacyIDEA 2.12
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: ok.

File: doc/installation/upgrade.rst [hunk 15]
```diff
@@ -73,6 +76,50 @@ Usually you will need to upgrade/migrate the database:
...
+change from your ppa sources to the new repositories. If you are upgrading your
+Ubuntu release, e.g. from 14.04 to 16.04 the principal steps are
+
+* Bring your Ubuntu 14.04 system up-to-date
+* Run the release upgrade (do-release-upgrade)
+* Eventually remove old repositories and add recent repositories as described in :ref:`add_ubuntu_repository`.
+* Reinstall/Upgrade privacyIDEA 3.x
+
+privacyIDEA 2.x installed the python packages to the system directly. The packages
+in the repository instead come with a virtual python environmentThis may cause lots
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: done

File: doc/configuration/realms.rst [hunk 16]
```diff
@@ -27,15 +27,15 @@ Relate User to a Realm
 
 There are several options to relate a user to a specific realm during
 authentication. Usually, if only a login name is given, the user will be
-searched in the default realm.
+searched in the default realm ``defrealm``.
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/configuration/realms.rst]: done

File: doc/configuration/tokenconfig/email.rst [hunk 17]
```diff
@@ -1,7 +1,7 @@
 
-.. _email_otp_token:
+.. _email_token_config:
 
-Email OTP Token
+Email OTP Token Configuration
 ...............
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/configuration/tokenconfig/email.rst]: done

File: doc/configuration/tokenconfig/email.rst [hunk 18]
```diff
@@ -35,46 +40,37 @@ sent to him in the response during the first step and only the OTP value. The
...
-
-If the mail server requires authentication you need to enter a username. If
-no username is entered, no authentication is performed on the mail server.
+**Concurrent Challenges**
 
-**Mail User Password**
+The config entry ``email.concurrent_challenges`` set in :ref:`cfgfile` will save the sent OTP
+value in the challenge database. This way several challenges can be open at the same
+time. The user can answer the challenges in an arbitrary order.
+Set this to a true value.
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/configuration/tokenconfig/email.rst]: done

File: doc/configuration/tokenconfig/sms.rst [hunk 19]
```diff
@@ -50,12 +50,13 @@ For configuring SMS Gateways read :ref:`sms_gateway_config`.
...
 **Concurrent Challenges**
 
-The config entry ``sms.concurrent_challenges`` will save the sent OTP
-value in the challenge database. This way several challenges can be open at the same
-time. The user can answer the challenges in an arbitrary order.
-Set this to a true value.
\ No newline at end of file
+If set to True in :ref:`cfgfile`, the config entry ``sms.concurrent_challenges``
+will save the sent OTP value in the challenge database. This way several challenges can
+be open at the same time. The user can answer the challenges in an arbitrary order.
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/configuration/tokenconfig/sms.rst]: done

File: doc/configuration/tokenconfig/email.rst (hunk 18 above)
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/configuration/tokenconfig/email.rst]: done

File: doc/installation/upgrade.rst [hunk 20]
```diff
@@ -14,11 +14,14 @@ system ways of *apt-get*, *aptitude* and *rpm* to upgrade privacyIDEA to the
 current version.
 
 If you want to upgrade your Ubuntu installation from privacyIDEA 2.23 to
-privacyIDEA 3.0, please read :ref:`upgrade_ubuntu`.
+privacyIDEA 3.0, please read :ref:`upgrade_packaged`.
 
-Basic pip upgrade process
+Basic upgrade process
 ~~~~~~~~~~~~~~~~~~~~~~~~~
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst [hunk 21]
```diff
@@ -14,11 +14,14 @@ system ways of *apt-get*, *aptitude* and *rpm* to upgrade privacyIDEA to the
...
 
+Upgrading a pip installation
+...............
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst [hunk 22]
```diff
@@ -36,7 +39,7 @@ and your database:
    pi-manage backup create
 
 Running upgrade
-...............
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: ok

File: doc/installation/upgrade.rst [hunk 23]
```diff
@@ -56,7 +59,7 @@ The following parameters are allowed:
 
 
 Manual upgrade
-..............
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: ok

File: doc/faq/rollout-strategies.rst [hunk 24]
```diff
@@ -30,6 +30,8 @@ While logged in, the user can enroll a token on his own.
 Automatic initial synchronization
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. todo:: What is autosync? (has incorrect reference)
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/faq/rollout-strategies.rst]: somehow the _autosync:: label has to be right before the headline to work.

File: doc/policies/index.rst [hunk 25]
```diff
@@ -69,6 +72,10 @@ Each policy can contain the following attributes:
   `string` and `integer` actions require an additional value - like
   ``scope=authentication:action='otppin=userstore'``.
 
+  .. note:: The number of supported policies increases gradually as sometimes special
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/policies/index.rst]: done

File: doc/policies/templates.rst [hunk 26]
```diff
@@ -37,4 +37,7 @@ A template must be referenced in a special ``index.json`` file::
    }
 
 where the key is the name of the template file and the value is a description
-displayed in the WebUI.
+displayed in the WebUI. A number of policies come preinstalled with privacyIDEA.
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/policies/templates.rst]: Added a sentence.

File: doc/tokens/index.rst [hunk 27]
```diff
@@ -0,0 +1,13 @@
+.. _tokens:
+
+Tokens
+------
+
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/tokens/index.rst]: added introduction

File: doc/tokens/supported_tokens.rst [hunk 28]
```diff
@@ -1,4 +1,6 @@
-Supported Tokens
+.. _supported_tokens:
+
+Hardware and Software Tokens
 ................
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/tokens/supported_tokens.rst]: :P

File: doc/webui/index.rst (hunk 4 above)
Comment (laclaro) [reply doc/webui/index.rst]: ok

File: doc/webui/index.rst (hunk 5 above)
Comment (laclaro) [reply doc/webui/index.rst]: removed

File: doc/webui/index.rst (hunk 6 above)
Comment (laclaro) [reply doc/webui/index.rst]: done

File: doc/webui/index.rst (hunk 7 above)
Comment (laclaro) [reply doc/webui/index.rst]: you wanted it...

File: doc/webui/index.rst (hunk 8 above)
Comment (laclaro) [reply doc/webui/index.rst]: Done. Also added a link to the plugins.

File: doc/tokens/tokentypes/questionnaire.rst (hunk 3 above)
Comment (laclaro) [reply doc/tokens/tokentypes/questionnaire.rst]: no config section there yet. Should we do one? I think it is not neccessary.

File: doc/tokens/tokentypes/questionnaire.rst (hunk 2 above)
Comment (laclaro) [reply doc/tokens/tokentypes/questionnaire.rst]: done. I added a note about the change of the aswers of existing tokens.

File: doc/tokens/tokentypes/pw.rst (hunk 1 above)
Comment (laclaro) [reply doc/tokens/tokentypes/pw.rst]: done.

File: doc/installation/ubuntu.rst [hunk 29]
```diff
@@ -7,12 +7,21 @@ Ubuntu Packages
...
 
 There are ready made packages for Ubuntu.
-Up to privacyIDEA 2.23 packages are available for 
-Ubuntu 14.04 LTS and 16.04 LTS [#ubuntu1604]_.
-These are available via a public ppa repository [#ppa]_.
 
-Starting with privacyIDEA 3.0 packages are available 
-for Ubuntu 16.04LTS and 18.04LTS not in PPA but in a new repository [#ubuntu1804]_.
+Packages of older releases of privacyIDEA up to version 2.23 are available for
+Ubuntu 14.04 LTS and Ubuntu 16.04 LTS from a public ppa repository [#ppa]_.
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/ubuntu.rst]: done

File: doc/webui/index.rst (hunk 4 above)
Comment (laclaro) [reply doc/webui/index.rst]: done

File: doc/webui/index.rst (hunk 5 above)
Comment (laclaro) [reply doc/webui/index.rst]: removed

File: doc/webui/index.rst (hunk 6 above)
Comment (laclaro) [reply doc/webui/index.rst]: added a sentence.

File: doc/webui/index.rst (hunk 7 above)
Comment (laclaro) [reply doc/webui/index.rst]: 1

File: doc/webui/index.rst (hunk 8 above)
Comment (laclaro) [reply doc/webui/index.rst]: done. I linkes to the plugins  section.

File: doc/installation/upgrade.rst (hunk 20 above)
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/upgrade.rst]: added instruction and links.

File: doc/installation/ubuntu.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/ubuntu.rst]: ok

File: doc/installation/ubuntu.rst [hunk 30]
```diff
@@ -66,33 +75,17 @@ or::
    deb http://lancelot.netknights.it/community/bionic/stable bionic main
 
 
-
-New installation of privacyIDEA 3.0
+Installation of privacyIDEA 3.0
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
[the remark this answers is not in the document]
Comment (laclaro) [reply doc/installation/ubuntu.rst]: ok

**Comment by laclaro** [comment]
Please review.

File: doc/application_plugins/index.rst [hunk 31]
```diff
@@ -21,6 +21,8 @@ various :ref:`authentication_modes`.
 Pluggable Authentication Module
 -------------------------------
 
+.. todo:: Note: privacyidea-pam is not in the ubuntu 18.04 repository.
```
Comment (cornelinux) [inline doc/application_plugins/index.rst]: @laclaro Yes. And it is not supposed to be there anymore.

**Pull request merged**